anndata/utils.py

```diff
@@ -65,10 +69,25 @@ def make_index_unique(index: pd.Index, join: str = "-"):
     values = index.values
```

Comment:
Could you make a copy of the values here? I was testing this and saw:

```python
b = a.copy()
b.obs_names_make_unique()
```

Could modify `a.obs_names`. This is because pandas assumes `pd.Index` objects are immutable, even though their underlying array is mutable. Another way to indirectly modify the index would be:

```python
obs_copy = a.obs.copy()
obs_copy.index.values[0] = "newval"
assert a.obs_names[0] == "newval"
```

Reply:
Ref for posterity: https://github.com/pandas-dev/pandas/issues/33001